src/QMCHamiltonians/NonLocalECPotential.deriv.cpp

```diff
@@ -137,26 +137,39 @@ NonLocalECPComponent::RealType NonLocalECPComponent::evaluateValueAndDerivatives
     for (int j = 0; j < nknot; j++)
     {
       PosType pos_now = W.R[iel];
+      //W.makeMoveAndCheck(iel, deltarV[j]);
+
+//#if defined(QMC_COMPLEX)
+//      psiratio[j] = psi.ratio(W, iel) * ValueType( std::cos(psi.getPhaseDiff()), std::sin(psi.getPhaseDiff()) );
+//#else
+//      psiratio[j] = psi.ratio(W, iel);
+//#endif
+//      psiratio[j] = psi.full_ratio(W, iel);
       W.makeMove(iel, deltarV[j]);
-#if defined(QMC_COMPLEX)
-      psiratio[j] = psi.ratio(W, iel) * std::cos(psi.getPhaseDiff());
-#else
-      psiratio[j] = psi.ratio(W, iel);
-#endif
+//#if defined(QMC_COMPLEX)
+//      psiratio[j] = psi.ratio(W, iel) * std::cos(psi.getPhaseDiff());
+//#else
+//      psiratio[j] = psi.ratio(W, iel);
+//#endif
+      psiratio[j] = psi.full_ratio(W, iel);
       psi.resetPhaseDiff();
 
       //use existing methods
       W.acceptMove(iel);
+      psi.acceptMove(W, iel);
```

Comment:
The call sequence matters.
W.makeMove
psi.ratio
psi.accept
W.accept

Reply:
changed.